src/effects/ObjectDetection.cpp

```diff
@@ -0,0 +1,285 @@
+                               (int)(bb_nrml.height*fh));
+                drawPred(detections.classIds.at(i), detections.confidences.at(i),
+                         box, cv_image);
+            }
+        }
+    }
+
+	// Set image with drawn box to frame
+    // If the input image is NULL or doesn't have tracking data, it's returned as it came
+	frame->SetImageCV(cv_image);
+
+	return frame;
+}
+
+void ObjectDetection::drawPred(int classId, float conf, cv::Rect2d box, cv::Mat& frame)
+{
+
+    //Draw a rectangle displaying the bounding box
+    cv::rectangle(frame, box, classesColor[classId], 2);
+
+    //Get the label for the class name and its confidence
+    std::string label = cv::format("%.2f", conf);
+    if (!classNames.empty())
+    {
+        CV_Assert(classId < (int)classNames.size());
+        label = classNames[classId] + ":" + label;
+    }
+
+    //Display the label at the top of the bounding box
+    int baseLine;
+    cv::Size labelSize = cv::getTextSize(label, cv::FONT_HERSHEY_SIMPLEX, 0.5, 1, &baseLine);
+
+    double left = box.x;
+    double top = std::max((int)box.y, labelSize.height);
+
+    cv::rectangle(frame, cv::Point(left, top - round(1.025*labelSize.height)), cv::Point(left + round(1.025*labelSize.width), top + baseLine), classesColor[classId], cv::FILLED);
+    putText(frame, label, cv::Point(left+1, top), cv::FONT_HERSHEY_SIMPLEX, 0.5, cv::Scalar(0,0,0),1);
+    std::cout<<"X1: "<<box.x<<" Y1: "<<box.y<<" X2: "<<box.width + box.x<<" Y2: "<<box.height + box.y<<"\n";
+    std::cout<<"Class: "<<label<<"\n\n";
+}
+
+// Load protobuf data file
+bool ObjectDetection::LoadObjDetectdData(std::string inputFilePath){
+    // Create tracker message
+    libopenshotobjdetect::ObjDetect objMessage;
+
+    {
+        // Read the existing tracker message.
+        fstream input(inputFilePath, ios::in | ios::binary);
+        if (!objMessage.ParseFromIstream(&input)) {
+            cerr << "Failed to parse protobuf message." << endl;
+            return false;
+        }
+    }
+
+    // Make sure classNames and detectionsData are empty
+    classNames.clear();
+    detectionsData.clear();
+
+    // Seed to generate same random numbers
+    std::srand(1);
+    // Get all classes names and assign a color to them
+    for(int i = 0; i < objMessage.classnames_size(); i++){
+        classNames.push_back(objMessage.classnames(i));
+        classesColor.push_back(cv::Scalar(std::rand()%205 + 50, std::rand()%205 + 50, std::rand()%205 + 50));
+    }
+
+    // Iterate over all frames of the saved message
+    for (size_t i = 0; i < objMessage.frame_size(); i++) {
+        // Create protobuf message reader
+        const libopenshotobjdetect::Frame& pbFrameData = objMessage.frame(i);
+
+        // Get frame Id
+        size_t id = pbFrameData.id();
+
+        // Load bounding box data
+        const google::protobuf::RepeatedPtrField<libopenshotobjdetect::Frame_Box > &pBox = pbFrameData.bounding_box();
```

Comment:
I've gotta say, these names are kind of hard on the eyes. What do you think of just `pb_objdetect`, `pb_tracker`, and `pb_stabilize`? 

(Commit da02180c79148ffcae27cfdddcfae10e10093799 in my fork actually already does that across the tree, so you don't have to go through all that if you decide to make the change. If not, totally fine by me. I just wanted to make sure there weren't any problems before I suggested it.)